### PR TITLE
FIX make the actions consistent on the grid field items to what they look like on pages

### PR DESCRIPTION
**Pull request by bergice**
Resolves https://github.com/silverstripe/silverstripe-admin/issues/904

### CODE REVIEW

File: src/VersionedGridFieldItemRequest.php [hunk 1]
```diff
@@ -270,22 +271,43 @@ protected function getRecordStatus()
      */
     protected function addVersionedButtons(DataObject $record, FieldList $actions)
     {
+        $stagesDiffer = $record->stagesDiffer();
+
+        // More options
+        $rootTabSet = new TabSet('ActionMenus');
+        $moreOptions = new Tab(
```
Comment (robbieaverill) [inline src/VersionedGridFieldItemRequest.php]:
```suggestion
        $moreOptions = Tab::create(
```

File: src/VersionedGridFieldItemRequest.php [hunk 2]
```diff
@@ -270,22 +271,43 @@ protected function getRecordStatus()
      */
     protected function addVersionedButtons(DataObject $record, FieldList $actions)
     {
+        $stagesDiffer = $record->stagesDiffer();
+
+        // More options
+        $rootTabSet = new TabSet('ActionMenus');
```
Comment (robbieaverill) [inline src/VersionedGridFieldItemRequest.php]:
```suggestion
        $rootTabSet = TabSet::create('ActionMenus');
```

File: src/VersionedGridFieldItemRequest.php [hunk 3]
```diff
@@ -270,22 +271,43 @@ protected function getRecordStatus()
...
+        $moreOptions = new Tab(
+            'MoreOptions',
+            _t(__CLASS__.'.MoreOptions', 'More options', 'Expands a view for more buttons')
```
Comment (robbieaverill) [inline src/VersionedGridFieldItemRequest.php]:
```suggestion
            _t(__CLASS__ . '.MoreOptions', 'More options', 'Expands a view for more buttons')
```

File: src/VersionedGridFieldItemRequest.php [hunk 4]
```diff
@@ -270,22 +271,43 @@ protected function getRecordStatus()
...
+                ->setAttribute('data-btn-alternate-remove', $noChangesClasses)
                 ->setUseButtonTag(true)
-                ->addExtraClass('btn btn-primary font-icon-rocket');
-
-            // Insert after save
-            if ($actions->fieldByName('action_doSave')) {
-                $actions->insertAfter('action_doSave', $publish);
-            } else {
-                $actions->push($publish);
+                ->setAttribute('data-text-alternate', _t(__CLASS__.'.BUTTONSAVEPUBLISH', 'Publish'))
```
Comment (robbieaverill) [inline src/VersionedGridFieldItemRequest.php]:
```suggestion
                ->setAttribute('data-text-alternate', _t(__CLASS__ . '.BUTTONSAVEPUBLISH', 'Publish'))
```

File: src/VersionedGridFieldItemRequest.php [hunk 5]
```diff
@@ -270,22 +271,43 @@ protected function getRecordStatus()
...
+            ;
+
+            // Set up the initial state of the button to reflect the state of the underlying SiteTree object.
+            if ($stagesDiffer) {
+                $publish->addExtraClass('btn-primary font-icon-rocket');
+                $publish->setTitle(_t(__CLASS__.'.BUTTONSAVEPUBLISH', 'Publish'));
```
Comment (robbieaverill) [inline src/VersionedGridFieldItemRequest.php]:
```suggestion
                $publish->setTitle(_t(__CLASS__ . '.BUTTONSAVEPUBLISH', 'Publish'));
```

**Comment by bergice** [comment]
@maxime-rainville should be fixed now.

**Pull request merged**